modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/resolver/ValueResolver.java

```diff
@@ -26,6 +29,19 @@
   T resolve(Event event) throws MuleException;
 
   /**
+   * Resolves a value from the given {@code event} and the {@link ConfigurationInstance config} associated
+   * to the component, if one is present.
+   *
+   * @param event a {@link Event}
+   * @param config an {@link Optional} {@link ConfigurationInstance}
+   * @return a resolved value
+   * @throws MuleException if the resolution of the value fails
+   */
+  default T resolve(Event event, Optional<ConfigurationInstance> config) throws MuleException {
```

Comment:
i don't like this method to begin with. But assuming you end up convincing me of it, it should be the other way around. `T resolve(Event event)` should have a default impl which calls this one with an empty optional. You should force ValueResolver developers to consider the possibility of the config being present.